### PR TITLE
Signup: update plans tables after AB test

### LINKED ISSUE #18660: Jetpack Connect: Plans page can have a horizontal scrollbar with Happychat opened
As @sirreal noticed in https://github.com/Automattic/wp-calypso/pull/18573#pullrequestreview-67920256, on smaller displays we can observe a horizontal scrollbar when we open Happychat in the plans page.

### PR DESCRIPTION
**Pull request by keoshi**
This PR removes the AB test code added in #23454 and updated later in #23802.
It also updates the existing code to always render the `mobileView` on smaller devices, and updates all associated styles.

More details in: p6TEKc-29X-p2

Fixes #18660

Related wp-e2e-tests PR: https://github.com/Automattic/wp-e2e-tests/pull/1271

### Testing

- Checkout this branch.
- Go to `/start/plans` or `/jetpack/connect/plans/[SITE_URL]` and verify that tables are presented correctly:
  - Desktop, responsive version for screens 661px wide or more.
  - Mobile, responsive version for screens 660px wide of less.

### CODE REVIEW

File: client/my-sites/plan-features/index.jsx [hunk 1]
```diff
@@ -97,10 +85,6 @@ class PlanFeatures extends Component {
 
 		mobileView = <div className="plan-features__mobile">{ this.renderMobileView() }</div>;
 
-		if ( isInSignup && abtest( 'mobilePlansTablesOnSignup' ) === 'original' ) {
-			mobileView = '';
-		}
-
 		return (
 			<div className={ planWrapperClasses } ref={ this.setScrollLeft }>
```
Comment (fditrapani) [inline client/my-sites/plan-features/index.jsx]: The `ref` here can also be removed.

**Comment by keoshi** [comment]
@fditrapani Thanks for reviewing! Pushed a commit to delete that unused method. As for the borders, I noticed that but it matches the existing ones on `/plans/`. I propose addressing this separately in a new PR.

**Comment by fditrapani** [comment]
> @fditrapani Thanks for reviewing! Pushed a commit to delete that unused method. As for the borders, I noticed that but it matches the existing ones on /plans/. I propose addressing this separately in a new PR.

Good call!

> Also, @fditrapani: to test it with a Jetpack site (or any changes for that matter), you can quick generate a site with https://jurassic.ninja/create

Wow, that's cool! I tried it out and it looks good!

File: client/my-sites/plan-features/style.scss [hunk 2]
```diff
@@ -445,29 +445,38 @@ $plan-features-sidebar-width: 272px;
...
 	}
 
 	.plan-features__table {
-		display: table !important;
-		border-spacing: 15px 0 !important;
 		width: 100%;
 		margin: 0;
+		padding: 0 16px;
+
+		@include breakpoint( "<660px" ) {
```
[the remark this answers is not in the document]
Comment (keoshi) [reply client/my-sites/plan-features/style.scss]: Yep, good catch. Pushed a commit that should sort this out.

**Comment by tyxla** [comment]
This is great 👍 💯

**Pull request merged**